lib/devShell.nix

```diff
@@ -0,0 +1,38 @@
+{ lib
+, mkShell
+, rustPlatform
+, cargo
+, rustc
+, pkgs
+}:
+
+{ checks ? {}
+, inputsFrom ? []
+, packages ? []
+, rust-analyzer ? pkgs.rust-analyzer
```

Comment:
I think we should avoid using `pkgs.rust-analyzer` here (instead hoist the binding to the top-level closure) so that we can pull the "correctly overlayed" version via `callPackage`.

(I really don't think anyone should be doing crazy stuff like `craneLib.overrideScope' (final: prev: { rust-analyzer = whatever; })` but in case there's shenanigans we won't make things worse)

I also anticipate it might have been written this way to avoid binding `rust-analyzer` to itself (even if the language supported it, it would look weird...). One suggestion I might provide is to not bind `rust-analyzer` here at all, but access it via another temporary variable:

```nix
let
  chosen-rust-analyzer = args.rust-analyzer or rust-analyzer;
in
lib.optional (chosen-rust-analyzer != null) chosen-rust-analyzer
```

Reply:
> I really don't think anyone should be doing crazy stuff like `craneLib.overrideScope' (final: prev: { rust-analyzer = whatever; })`

Doesn't seem _that_ far-fetched, `rust-analzyer` moves quickly enough that people might want to get it from an overlay or a separate flake or something.

Reply:
I hoisted the definition to the top-level binding — is that sufficient for getting this to work with overlays and overrides?